### PR TITLE
[Refactor] Rename table property replicated_storage to single_leader_replication

### PR DESCRIPTION
**Pull request by meegoo**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [x] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

### CREATE TABLE
```
CREATE TABLE my_table
(
    k1 TINYINT,
    k2 DECIMAL(10, 2) DEFAULT "10.5",
    v1 CHAR(10) REPLACE,
    v2 INT SUM
)
ENGINE = olap
AGGREGATE KEY(k1, k2)
DISTRIBUTED BY HASH(k1) BUCKETS 10
PROPERTIES ("single_leader_replication" = true);
```
### ALTER TABLE
```
ALTER TABLE my_table
SET ("single_leader_replication" = true);
```

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by meegoo** [comment]
run starrocks_fe_unittest

**Comment by meegoo** [comment]
run starrocks_fe_unittest

File: fe/fe-core/src/main/java/com/starrocks/common/Config.java [hunk 1]
```diff
@@ -1138,7 +1138,7 @@ public class Config extends ConfigBase {
      * enable replicated storage as default table engine
      */
     @ConfField(mutable = true)
-    public static boolean enable_replicated_storage_as_default_engine = true;
+    public static boolean enable_single_leader_replication_as_default = true;
```
Comment (chaoyli) [inline fe/fe-core/src/main/java/com/starrocks/common/Config.java:1141]: Why to change the name?

**Comment by howrocks** [comment]
This pr can be closed  @meegoo.

We will not  change this config name finally because the potential misunderstandging of the new name, please know it. @evelynzhaojie @jaogoy

**Pull request closed without merging**